app/src/main/java/com/rockthevote/grommet/ui/MainActivityViewModel.kt

```diff
@@ -90,6 +87,26 @@ class MainActivityViewModel(
 
     private suspend fun loadRequestsFromDb() = registrationDao.getAll()
 
+    /**
+     * Asynchronously determines if the user can clock out. Calls [successCallback]
+     * when the user can logout, [failCallback] when the user cannot.
+     *
+     * Note: Coroutines cannot be used from Java, so callbacks are necessary
```

Comment:
The primary consumer here is a View class, so it was easier to use callbacks than observe the ViewModel state.